#N8c834Zu Sign In users

Ntwali-Josue opened the pull request:
**What does this PR do**
This PR contains a backend work that allows users of phantom to login to the platform.

**Description of tasks to be completed**
The user should be able to login to the app by providing their email & password.
The API should return a JW token on a successful login.

**how should this be manually tested**
run `https://eagles-phantom-backend.herokuapp.com/v1/auth/login` in postman

**Other related stories**
https://trello.com/c/N8c834Zu/8-sign-in-to-phantom

**screenshot**
![image](https://user-images.githubusercontent.com/58233753/95759493-0dc63800-0caa-11eb-928d-5876952f2bfe.png)

**Addition**
Access swagger documentation by running :
`https://eagles-phantom-backend.herokuapp.com/api-docs/`

Ally4 reviewed (commented):
it`s not woks, no matter what I`m putting is it tells me the wrong email

Ntwali-Josue commented:
@Ally4 you need to use the saved email which is Josh@gmail.com and the password admin

Ally4 commented:
I would like to ask you to edit the title of the PR too

Ntwali-Josue committed:
Update README.md

fix(heroku): heroku pipeline

deleting and putting .nyc_output in gitignore

[(finishes) #1oYL5X1g]

Update README.md

Update README.md

Update README.md

fix(travis): solve travis error

ceelogre commented on src/database/models/index.js:
```diff
@@ -0,0 +1,35 @@
+const fs = require('fs');
+const path = require('path');
+const Sequelize = require('sequelize');
+const envConfigs = require('../config/config');
+
+const basename = path.basename(__filename);
+const env = process.env.NODE_ENV || 'development';
```
Is there any particular reason this is using `require` ?

Ntwali-Josue replied:
Because the models are created automatically, they come in es5

ceelogre commented on src/languages/locales/fr.json:
```diff
@@ -1,4 +1,7 @@
-    "Welcome to phantom!.":"bienvenue chez phantom!."
+    "Welcome to phantom!.":"bienvenue chez phantom!.",
+    "logged In successfull": "Connexion réussie",
+	"wrong email or password": "mauvais email ou mot de passe",
+	"Invalid email or password": "email ou mot de passe invalide"
```
These can be merged into one message.

ceelogre commented on src/test/test.js:
```diff
@@ -1,19 +1,31 @@
+        expect(res.body.message).to.equals('wrong email or password');
         done();
       });
   });
```
Add (at least) one more test for the case where a user provides the right email and password.

ceelogre reviewed (changes requested):
1. Check out the comments I've added
2. Add the right `Accept Language`
3. Show log messages only in `dev` mode.

ceelogre reviewed (changes requested):
This pr doesn't have `.env.example`
